addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.evohome.internal.api.EvohomeApiClient;
+import org.openhab.binding.evohome.internal.api.EvohomeApiClientV2;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Gateway;
+import org.openhab.binding.evohome.internal.api.models.v2.response.GatewayStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Location;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Locations;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationsStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystem;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystemStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Zone;
+import org.openhab.binding.evohome.internal.api.models.v2.response.ZoneStatus;
+import org.openhab.binding.evohome.internal.configuration.EvohomeAccountConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides the bridge for this binding. Controls the authentication sequence.
+ * Manages the scheduler for getting updates from the API and updates the Things it contains.
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class EvohomeAccountBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EvohomeAccountBridgeHandler.class);
+    private EvohomeAccountConfiguration configuration;
+    private EvohomeApiClient apiClient;
+    private List<AccountStatusListener> listeners = new CopyOnWriteArrayList<AccountStatusListener>();
+
+    protected ScheduledFuture<?> refreshTask;
+
+    public EvohomeAccountBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(EvohomeAccountConfiguration.class);
+
+        if (checkConfig()) {
+            try {
+                apiClient = new EvohomeApiClientV2(configuration);
+
+                // Initialization can take a while, so kick if off on a separate thread
+                scheduler.schedule(new Runnable() {
```

Comment:
you could use lambda syntax: `scheduler.schedule(() -> {...}, 0, TimeUnit.SECONDS);`